src/main/java/javax/jmdns/impl/ServiceInfoImpl.java

```diff
@@ -348,7 +348,6 @@ public ServiceInfoImpl(Map<Fields, String> qualifiedNameMap, int port, int weigh
             instance = "";
             // throw new IllegalArgumentException("The instance name component of a fully qualified service cannot be empty.");
         }
-        instance = removeSeparators(instance);
```

Comment:
This line looks as if it was there for some important reason. Simply removing it might solve the one issue, but cause other regressions. Could you elaborate why you think it is not needed (or rather, did you understand why it was there in the first place)?